flixel/util/FlxCollision.hx

```diff
@@ -98,12 +99,17 @@ class FlxCollision
 		matrixB.identity();
 		matrixB.translate(-(intersect.x - boundsB.x), -(intersect.y - boundsB.y));
 		
+	#if flash
+		if (Contact._frame.sourceSize.x != Contact.framePixels.width || Contact._frame.sourceSize.y != Contact.framePixels.height
+		|| Target._frame.sourceSize.x != Target.framePixels.width || Target._frame.sourceSize.y != Target.framePixels.height)
+	#else
 		if (FlxG.renderTile)
+	#end
```

Comment:
we don't need both checks:
```
#if flash
		if (Contact._frame.sourceSize.x != Contact.framePixels.width || Contact._frame.sourceSize.y != Contact.framePixels.height || Target._frame.sourceSize.x != Target.framePixels.width || Target._frame.sourceSize.y != Target.framePixels.height)
 #else
  		if (FlxG.renderTile)
 #end
```

so we just call `Contact.drawFrame();` and `Target.drawFrame();`.

`drawFrame()` (and following methods which are called by it) have checks for value of `dirty` flag, so if sprite doesn't need to update its frame graphic then it will skip this step.
The only perfomance problem is with animated sprites when animation could have series of the same frames: when animation changes the frame (even if it has the same index) then it automatically resets sprite's frame, whih set dirty to true